doc/heps/hep2.md

```diff
@@ -0,0 +1,370 @@
+# HEP 2: Release and deprecation policies
+
+<table>
+<tr><td> Identifier </td><td> HEP 2 </td>
+<tr><td> Title </td><td> Release and deprecation policies </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Maxime Liquet </td></tr>
+<tr><td> Created </td><td> TBD </td></tr>
+<tr><td> Updated </td><td> - </td></tr>
+<tr><td> Discussion </td><td> https://github.com/holoviz/holoviz/pull/388 </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Summary
+
+This HEP describes the release and deprecation policies adopted by the HoloViz Projects. While two HEPs could have be written to describe these policies separately, it is deemed that they are enough intertwined to be addressed together (e.g. to answer questions such as in which kind of final release can a feature be deprecated/removed?).
+
+The Projects already follow a pretty consistent procedure when it comes to releases, which this HEP formalizes. However, the Projects have applied various approaches for their deprecation cycle, and without clear guidelines this has sometimes led to long and unproductive discussions. The HEP aims to improve this by introducing new and well defined guidelines. Making it easier for Projects to deprecate features is **key to their long-term maintenance**.
+
+The overall goal of the HEP is to ensure that these policies are **consistently applied across the Projects**, which will help provide a **consistent user experience** and **help contributors and maintainers make decisions**. We also aim to adopt **standard practices**.
+
```

Comment:
Let's add a concise 'Resolution' section after 'Summary' for a quick recap of the technical plan of action. Something like:

```
## Resolution
- Adopt a tiered warning strategy for deprecations:
  - Initially emit a `DeprecationWarning`, ensuring it's set with the appropriate stacklevel for accurate source identification.
  - After a 12-month period, elevate the warning to a `FutureWarning` to capture the attention of all users, including those who might not be configured to see DeprecationWarnings.
- Allow exceptions in rare cases, such as beginning with a `PendingDeprecationWarning` for potentially reversible or highly disruptive changes, or immediately using a `FutureWarning` for urgent cases.
```

Reply:
I don't really want to summarize the plan of action as I think all the steps of the deprecation cycle are important.